run

```diff
@@ -22,21 +22,34 @@ parser.add_argument(
     action="store_true",
     help="Run container in background & don't show container logs",
 )
+# Set to false if your current user is already in the docker user group
+parser.add_argument(
+    "--no-sudo",
+    default=False,
+    action="store_true",
+    help="Run commands without sudo",
+)
 args = parser.parse_args()
-
 if not os.path.exists("node_modules"):
     print("Node modules have not been installed yet, doing so now")
     subprocess.call(["npm", "install"])
 
 print("Stopping {} container (if running)".format(CONTAINER_NAME))
+remove_container_cmd = ["sudo", "docker", "rm", "--force", CONTAINER_NAME]
+if args.no_sudo:
+    remove_container_cmd.remove("sudo")
+
 subprocess.call(
-    ["sudo", "docker", "rm", "--force", CONTAINER_NAME],
+    remove_container_cmd,
     stdout=subprocess.DEVNULL,
     stderr=subprocess.STDOUT,
 )
 
 print("Building container")
-subprocess.check_call(["sudo", "docker", "build", ".", "-t", IMAGE_NAME])
+build_container_cmd = ["sudo", "docker", "build", ".", "-t", IMAGE_NAME]
+if args.no_sudo:
+    build_container_cmd.remove("sudo")
```

Comment:
Given that this pattern is repeated you could clean things up by having a function which optionally adds sudo. 

If you wanted to go even further and you could define a function dynamically like this depending on the option:

```python
parser.add_argument(
    "--no-sudo",
    default=False,
    action="store_true",
    help="Run commands without sudo",
)
args = parser.parse_args()


if args.no_sudo:
   add_sudo = lambda cmd: cmd
else:
   add_sudo = lambda cmd: ["sudo"] + cmd
```

Then later the code that constructs the commands just end up looking like...
```python
build_container_cmd = add_sudo(["docker", "build", ".", "-t", IMAGE_NAME])
```

No ifs required, less lines of code, less chance for errors.

Reply:
Have implemented